src/test-table.md

```diff
@@ -100,6 +104,87 @@ for _, tt := range tests {
 }
 ```
 
+### Avoid Messy Table Tests
+
+Table tests become difficult to read and maintain if there becomes significant
+logic forks within the subtest `for` loop. Table tests should **NOT** be used whenever 
+there needs to be complex logic added inside this execution loop.
```

Comment:
nit:

- s/if there becomes significant/if there are significant/
- s/within the subtest `for`/within the subtest's `for`/

Alternatively:

```suggestion
Table tests can be difficult to read or maintain if their subtests contain
conditional assertions or other branching logic. Table tests should **NOT** be
used whenever there needs to be complex or conditional logic inside subtests.
```